hbase-server/src/main/java/org/apache/hadoop/hbase/master/cleaner/CleanerChore.java

```diff
@@ -322,140 +330,154 @@ public boolean getEnabled() {
     T act() throws IOException;
   }
 
-  /**
-   * Attemps to clean up a directory, its subdirectories, and files. Return value is true if
-   * everything was deleted. false on partial / total failures.
-   */
-  private final class CleanerTask extends RecursiveTask<Boolean> {
-    private static final long serialVersionUID = -1584635903138015418L;
-    private final Path dir;
-    private final boolean root;
+  private interface Callback {
+    void run(boolean val);
+  }
 
-    CleanerTask(final FileStatus dir, final boolean root) {
-      this(dir.getPath(), root);
+  private final class AsyncResult {
```

Comment:
A thought, can `AsyncResult` be inside `DirScanPool` rather than `CleanerChore`?

Reply:
IMO,  the DirScanPool is mainly used for thread pool managing, such as control pool size, submit runnable task etc.  The  AsyncResult is used for getting the result async in the upper layer, so I think we'd better to keep it in CleanChore.
Thanks.